### DISCUSSION TITLE
Problem with print out after running doPriceCut in wedding planer problem

### DISCUSSION THREAD

**Discussion by opperud** [coin-or/Dip#138] [Q&A]
Hey!

I'm playing around with the wedding planner problem, and trying out different branch price and cut strategies to solve the problem. The code itself works, and reaches the optimal solution quite fast. However, the problem that I have encountered is that I can not print anything after I have called the Solve function, and it is therefore impossible to see what the optimal solution consist of in terms of variable values. This only happens when I add the dippyOpts['doPriceCut'] = '1'. 

The full code in DiPPY can be found below. 

from __future__ import print_function
from builtins import range
import pulp
import argparse
import math
from past.utils import old_div
from coinor.dippy import DipProblem, DipSolStatOptimal, Solve
from coinor.dippy.examples.gen_func import *

debug_print = False
max_table_size = 4

guests = 'A B C D E F G H I J K L M N O P Q R S T U V W X Y Z'.split()

max_tables = math.ceil(len(guests)/max_table_size)

tol = pow(pow(2, -24), old_div(2.0, 3.0))

def happiness(guest_a, guest_b):
    """
    Return the happiness (0 is the best) of allocating two
    guests together in the same table
    """
    return abs(ord(guest_a) - ord(guest_b))

parser = argparse.ArgumentParser(description='Solve a wedding planner problem.')
addDippyArgs(parser)
args = parser.parse_args()

#create the set of possible tables
tables = list(range(max_tables))

possible_seatings = [(g, t) for g in guests
                            for t in tables]

#create a binary variable to model if a guest sits at a particular table
x = pulp.LpVariable.dicts('possible_seatings', possible_seatings, 
                            lowBound = 0,
                            upBound = 1,
                            cat = pulp.LpInteger)

seating_model = DipProblem("Wedding_Seating_Model_(DIP)", pulp.LpMinimize)

#specify the maximum number of guests per table
for table in tables:
    seating_model.relaxation[table] += sum([x[(guest, table)]
                                            for guest in guests]) <= \
                                            max_table_size, \
                                            "Maximum_table_size_%s"%table

#A guest must seated at one and only one table
for guest in guests:
    seating_model += (sum([x[(guest, table)] for table in tables]) == 1, 
                      "Must_seat_%s"%guest)

#create a set of variables to model the objective function
possible_pairs = [(a, b) for a in guests for b in guests if ord(a) < ord(b)]
happy = pulp.LpVariable.dicts('table_happiness', tables,
                              lowBound = 0,
                              upBound = None,
                              cat = pulp.LpContinuous)

seating_model += sum([happy[table] for table in tables])

#create constraints for each possible pair
for table in tables:
    for (a, b) in possible_pairs:
        seating_model.relaxation[table] += \
            happy[table] >= (happiness(a, b) * (x[(a, table)] + 
                                                x[(b, table)] - 1))

def relaxed_solver(prob, table, redCosts, target):
    """
    Generate columns (tables) with negative reduced costs
    """
    dvs = []
    neg_guests = [g for g in guests
                       if redCosts[x[(g,table)]] < 0.0]
    neg_guests.sort()
    # find all possible tables between two end points
    for pos1, pos2 in [(i, j) for i in range(len(neg_guests))
                            for j in range(len(neg_guests))
                                if j > i]:
        # find the suitable guests that can be included in between the end 
        # points
        candidate_guests = [(redCosts[x[(g,table)]], g)
                                    for g in neg_guests[pos1+1:pos2]]
        candidate_guests.sort()
        # pick the best guests (ie those with the negative reduced costs)
        possible_table_inner = [g 
                            for _, g in candidate_guests[:max_table_size-2]]
        #This is the best table between the end points
        possible_table = [neg_guests[pos1]] + possible_table_inner +\
                            [neg_guests[pos2]]
        # calculate the sum of the reduced costs for each of the guests
        neg_cost = sum(redCosts[x[(g, table)]] for g in possible_table)
        table_happiness = happiness(possible_table[0], possible_table[-1])
        rc = neg_cost + table_happiness * redCosts[happy[table]]
        var_values = [(x[(g, table)], 1) 
                      for g in possible_table]
        var_values.append((happy[table], table_happiness))
        dvs.append(dict(var_values))
        if debug_print:
            print('Table: ', table, 'Happiness: ', table_happiness, 'RC: ', rc)

    return DipSolStatOptimal, dvs

seating_model.relaxed_solver = relaxed_solver

seating_model.writeLP('wedding_main.lp')
for table in tables:
   seating_model.writeRelaxed(table, 'wedding_relax%s.lp' % table);

tol = pow(pow(2, -24), old_div(2.0, 3.0))

dippyOpts = addDippyOpts(args)
dippyOpts['TolZero'] = str(tol)
dippyOpts['generateInitVars'] = '1'
dippyOpts['doPriceCut'] = '1'
dippyOpts['CutCGL'] = '1'

Solve(seating_model, dippyOpts)

for table in tables:
    print(table, end=' ')
    for guest in guests:
        if x[(guest,table)].value() >= 0.99:
            print(guest, end=' ')
    print(happy[table].value())

**Comment by tkralphs** [comment]
I just wanted to say that I saw this, but haven't gotten a chance to look at it yet. I'll try to do so as soon as possible.
